### PR TITLE
add ip to grant claim and forward to grant server

### PR DESCRIPTION
**Pull request by evq**

### CODE REVIEW

File: bat-utils/lib/extras-hapi.js [hunk 1]
```diff
@@ -154,7 +154,7 @@ const WreckProxy = (server, opts) => {
 
   if (!opts) opts = {}
   if (!opts.headers) opts.headers = {}
-  if (!opts.headers['user-agent']) opts.headers['user-agent'] = wreckUA
+  if (!opts.headers['user-agent'] && !opts.headers['User-Agent']) opts.headers['user-agent'] = wreckUA
```
Comment (mrose17) [inline bat-utils/lib/extras-hapi.js]: i do't object to the change, but what caused you to make the change? just curious.

File: bat-utils/lib/hapi-auth-whitelist.js [hunk 2]
```diff
@@ -27,12 +27,25 @@ exports.authorizedP = (ipaddr) => {
...
 exports.ipaddr = (request) => {
-  return (request.headers['x-forwarded-for'] || request.info.remoteAddress).split(',')[0].trim()
+  // https://en.wikipedia.org/wiki/X-Forwarded-For    X-Forwarded-For: client, proxy1, proxy2
+  // Since it is easy to forge an X-Forwarded-For field the given information should be used with care.
+  // The last IP address is always the IP address that connects to the last proxy, which means it is the most reliable source of information.
+
+  const forwardedFor = request.headers['x-forwarded-for']
+  if (forwardedFor) {
+    const forwardedIps = forwardedFor.split(',')
+    return forwardedIps[forwardedIps.length - 1].trim() || request.info.remoteAddress
```
Comment (mrose17) [inline bat-utils/lib/hapi-auth-whitelist.js]: in the future, probably more consistent to use `underscore.last(forwardedFor.split(',').trim()`

File: bat-utils/lib/extras-hapi.js (hunk 1 above)
Comment (mrose17) [reply bat-utils/lib/extras-hapi.js]: nvm... it's `go`

Comment (evq) [reply bat-utils/lib/extras-hapi.js]: I seemed to only be able to override the outgoing ip address by using `User-Agent` versus `user-agent` - but don't hold me to it :)

**Pull request merged**